pulp_smash/tests/pulp3/pulpcore/api_v3/test_tasks.py

```diff
@@ -0,0 +1,88 @@
+from pulp_smash.tests.pulp3.pulpcore.utils import gen_repo
+from pulp_smash.tests.pulp3.utils import get_auth
+from pulp_smash.tests.pulp3.pulpcore.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+_DYNAMIC_TASKS_ATTRS = ('finished_at',)
+"""Task attributes that are dynamically set by Pulp, not set by a user."""
+
+
+class TasksTestCase(unittest.TestCase, utils.SmokeTest):
+    """Perform different operation over tasks.
+
+    This test targets the following issues:
+
+    * `Pulp #3144 <https://pulp.plan.io/issues/3144>`_
+    * `Pulp #3527 <https://pulp.plan.io/issues/3527>`_
+    * `Pulp Smash #754 <https://github.com/PulpQE/pulp-smash/issues/754>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.client = api.Client(config.get_config(), api.json_handler)
+        cls.client.request_kwargs['auth'] = get_auth()
+        cls.task = {}
+
+    def test_01_create_task(self):
+        """Create a task."""
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo['_href'])
+        attrs = {'description': utils.uuid4()}
+        response = self.client.patch(repo['_href'], attrs)
+        self.task.update(self.client.get(response['_href']))
+
+    @selectors.skip_if(bool, 'task', False)
+    def test_02_read_href(self):
+        """Read a task by its _href."""
+        task = self.client.get(self.task['_href'])
+        for key, val in self.task.items():
+            if key in _DYNAMIC_TASKS_ATTRS:
+                continue
+            with self.subTest(key=key):
+                self.assertEqual(task[key], val, task)
+
+    @selectors.skip_if(bool, 'task', False)
+    def test_02_read_invalid_worker(self):
+        """Read a task using an invalid worker name."""
+        with self.assertRaises(HTTPError):
+            self.filter_tasks({'worker': utils.uuid4()})
+
+    @selectors.skip_if(bool, 'task', False)
+    def test_02_read_valid_worker(self):
+        """Read a task using a valid worker name."""
+        page = self.filter_tasks({'worker': self.task['worker']})
+        self.assertNotEqual(len(page['results']), 0, page['results'])
+
+    def test_02_read_invalid_date(self):
+        """Read a task by an invalid date."""
+        page = self.filter_tasks({
+            'finished_at': utils.uuid4(),
+            'started_at': utils.uuid4()})
+        self.assertEqual(len(page['results']), 0, page['results'])
+
+    @selectors.skip_if(bool, 'task', False)
+    def test_02_read_valid_date(self):
+        """Read a task by a valid date."""
+        page = self.filter_tasks({'started_at': self.task['started_at']})
+        self.assertGreaterEqual(len(page['results']), 1, page['results'])
+
+    @selectors.skip_if(bool, 'task', False)
+    def test_03_delete_tasks(self):
+        """Delete a task."""
+        if self.task['state'] in (
+                'canceled', 'completed', 'failed', 'skipped'):
```

Comment:
Rather than hard-coding this constant here, how about referencing the existing constant?

Also, `self.task` should already be in a completed state. For it to be otherwise would indicate a problem either in Pulp's tasking system or Pulp Smash's logic for interacting with it. How about code like this?

```python
    @selectors.skip_if(bool, 'task', False)
    def test_03_delete_tasks(self):
        """Delete a task."""
        # If this assertion fails, then either Pulp's tasking system or Pulp
        # Smash's code for interacting with the tasking system has a bug.
        self.assertIn(self.task['state'], _P3_TASK_END_STATES)
        self.client.delete(self.task['_href'])
        with self.assertRaises(HTTPError):
            self.client.get(self.task['_href'])
```